datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1753,6 +1754,23 @@ pub enum Partitioning {
 }
 
 /// Unnest a column that contains a nested list type.
+///
+/// For example, calling unnest(c1) results in the following:
```

Comment:
the current implementation adds a row if there's a null value and would likely skip the row if the array is empty.

I can work on fixing this if that's needed

Reply:
Yes when I implemented `unnest_column` I followed the logic in Pandas and Polars:

```
          ints  shape_id
[3, 88, 94]         1
       [73]         2
       None         3
   [43, 97]         4
       None         5
       [65]         6
```

by unnesting `ints` we get:

```
   ints  shape_id
   3         1
  88         1
  94         1
  73         2
None         3
  43         4
  97         4
None         5
  65         6
```

that I think is better because you preserve all the information that was in the nested data.

Reply:
If you decide to implement the same behavior you have in DuckDB would be nice to have an option to preserve the old behavior.

Reply:
I have the code ready to change this but need someone to confirm if this is what we want. 
To support both ways, we could introduce some extra boolean flag like `keep_nulls`.

Reply:
I think a `keep_nulls` flag sounds like a good idea to me

So is the consensus for "duckdb/clickhouse" behavior by default, with a flag for `keep_nulls`? If so, I can update the examples in this PR

Reply:
Maybe to avoid breaking API changes to the `LogicalPlanBuilder::unnest_column` interface we can add a `LogicalPlanBuilder::unnest_with_options` method that will set the flag in the `LogicalPlan::Unnest` struct.

Reply:
I'll have a play with the API and make a proposal

Reply:
Here is a proposed API -- please let me know what you think: https://github.com/apache/arrow-datafusion/pull/7088#